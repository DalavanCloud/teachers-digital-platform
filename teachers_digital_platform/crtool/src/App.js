import React, { Component } from 'react';
import './App.css';

import CustomerReviewToolComponent from "./js/components/CustomerReviewToolComponent";

class App extends Component {
  render() {
    return (
      <div className="App">
        <CustomerReviewToolComponent />
      </div>
    );
  }
}

export default App;
