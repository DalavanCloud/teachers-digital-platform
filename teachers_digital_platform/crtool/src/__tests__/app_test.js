import React from 'react';
import ReactDOM from 'react-dom';
import App from '../App';
import localStorage from 'mock-local-storage';
import renderer from 'react-test-renderer';

describe ('initial state', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  
})