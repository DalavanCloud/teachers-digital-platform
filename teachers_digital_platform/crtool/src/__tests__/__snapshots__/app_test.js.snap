// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="App"
>
  <div
    className="l-survey-top"
  >
    <button
      className="a-btn a-btn__link"
      onClick={[Function]}
    >
      <span
        className="a-icon a-icon__large"
      >
        <svg
          class="cf-icon-svg"
          viewBox="0 0 1000 1200"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M500 105.2c-276.1 0-500 223.9-500 500s223.9 500 500 500 500-223.9 500-500-223.9-500-500-500zm59.5 816.2c-6.5 15.2-18.6 27.3-33.7 33.7l-.3.1c-7.7 3.2-15.9 4.9-24.2 4.9-25.8.2-49.2-15.1-59.3-38.9-10.2-23.7-4.8-51.2 13.6-69.3 6-5.7 12.9-10.3 20.6-13.4 7.9-3.3 16.5-5 25.1-5 8.3 0 16.5 1.7 24.2 5l.2.1c7.5 3.2 14.3 7.7 20.2 13.4 18.3 18.2 23.6 45.6 13.6 69.4zm115.7-430.5c-4.1 15.2-10.1 29.9-17.7 43.7-7 12.4-15.1 24.2-24.2 35.1-8.8 10.4-17.3 20.3-25.4 29.5-.4.4-.8.8-1.2 1.3-7.6 8-14.6 15.6-20.8 22.9-6.6 7.6-12.4 15.9-17.4 24.7-5.2 9.2-9.1 19.1-11.8 29.3-2.8 10.7-4.3 23.2-4.3 37.2v2.2c0 27.4-22.4 49.8-50 49.8s-50-22.4-50-50v-2.2c0-22.7 2.6-43.8 7.6-62.9 4.9-18.6 12.1-36.4 21.6-53.2 8.2-14.3 17.7-27.8 28.5-40.3 7.1-8.3 15.1-17 23.6-26 7.3-8.3 15.1-17.3 23.1-26.8 5.2-6.3 9.8-13 13.8-20.1 3.6-6.7 6.4-13.8 8.4-21.2 1.4-5.1 3.1-14.1 3.1-28.4a80.7 80.7 0 0 0-6.2-32.5c-4.1-10.1-10.1-19.2-17.7-27-7.5-7.6-16.5-13.7-26.4-17.8-9.7-4.2-20.1-6.2-30.7-6-10.9-.2-21.7 1.9-31.7 6.2-20.5 8.5-36.7 24.8-45.2 45.3-4.3 10-6.4 20.9-6.2 31.8v1.2c0 27.6-22.4 50-50 50s-50-22.4-50-50v-1.3c0-25.5 4.7-48.8 14.3-71.2 18.5-44 53.5-79 97.4-97.6 22.4-9.7 45.8-14.4 71.3-14.4 24.2-.2 48.3 4.7 70.5 14.4 43.7 18.7 78.3 53.7 96.4 97.6 9.5 22.5 14.1 45.8 14.1 71.4 0 20.5-2.3 39.1-6.8 55.3z"
          />
        </svg>
      </span>
        Can I save my work?
    </button>
    <div
      aria-describedby="modal-save-work_desc"
      aria-hidden="true"
      aria-labelledby="modal-save-work_title"
      className="o-modal"
      id="modal-save-work"
      role="alertdialog"
    >
      <div
        className="o-modal_backdrop"
      />
      <div
        className="o-modal_container"
      >
        <form
          className="o-modal_content"
        >
          <span>
            <div
              className="o-modal_body"
            >
              <button
                className="o-modal_close a-btn a-btn__link"
                onClick={[Function]}
              >
                Close  
                <span
                  className="a-icon a-icon__large"
                >
                  <svg
                    className="cf-icon-svg"
                    viewBox="0 0 1000 1200"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M500 105.2c-276.1 0-500 223.9-500 500s223.9 500 500 500 500-223.9 500-500-223.9-500-500-500zm261.8 692.2c19.4 19.6 19.3 51.3-.3 70.7-19.5 19.3-50.9 19.3-70.4 0L499.6 676.6 308 868.1c-19.6 19.4-51.3 19.3-70.7-.3-19.3-19.5-19.3-50.9 0-70.4l191.6-191.5-191.6-191.6c-19.3-19.8-18.9-51.4.9-70.7 19.4-18.9 50.4-18.9 69.8 0l191.6 191.5 191.5-191.5c19.6-19.4 51.3-19.3 70.7.3 19.3 19.5 19.3 50.9 0 70.4L570.3 605.9l191.5 191.5z"
                    />
                  </svg>
                </span>
              </button>
              <h1
                id="modal-save-work_title"
              >
                Saving your work
              </h1>
              <div
                id="modal-save-work_desc"
              >
                <p>
                  This tool uses cookies to 
                  <strong>
                    temporarily
                  </strong>
                   save your work. To see answers you’ve already completed, you need to use the same computer and browser, and don’t clear your cookies.
                </p>
                <p>
                  To save a permanent copy of your work, please print the summary or save it as a PDF for every dimension as you complete it. You can also print or save a summary of the entire review. Learn how to 
                  <a
                    href="https://www.consumerfinance.gov/consumer-tools/save-as-pdf-instructions/"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    save the summary as a PDF
                  </a>
                  .
                </p>
                <p>
                  You can only work on a single curriculum at a time
                </p>
              </div>
            </div>
            <div
              className="o-modal_footer"
            >
              <button
                className="a-btn"
                onClick={[Function]}
              >
                Close
              </button>
            </div>
          </span>
        </form>
      </div>
    </div>
  </div>
  <div
    class="h5 u-mb30"
  >
    You’re reviewing
  </div>
  <h1 />
  <ul
    className="o-dimension-selection-bar"
  >
    <li
      className="o-dimension-section-bar_dimension"
    >
      <button
        className="o-dimension-section-bar_button CONTENT"
        onClick={[Function]}
      >
        <div
          class="o-dimension-section-bar_selector"
        >
          <span
            className="o-dimension-section-bar_selector-title"
          >
            <span
              className="a-icon a-icon__large"
            >
              <svg
                class="cf-icon-svg"
                viewBox="0 0 651.7 1200"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M30 161c-16.5 0-30 13.5-30 30v827.8c0 16.5 13.5 30 30 30h591.7c16.5 0 30-13.5 30-30V343.7L469 161H30zm389.6 60v134.8c0 19.9 16.3 36.2 36.2 36.2h135.9V988.8H60V221h359.6z"
                />
                <path
                  d="M123.8 768.6h394.8v50H123.8zM123.8 644h394.8v50H123.8zM123.8 519.5h394.8v50H123.8z"
                />
                <circle
                  cx="194"
                  cy="382.3"
                  r="60"
                />
              </svg>
            </span>
             
            Content
             
            <span
              className="o-dimension-section-bar_selector-status"
            >
               
            </span>
          </span>
          <span
            className="o-dimension-section-bar_selector-text"
          >
            Covers core knowledge and skills in national standards
          </span>
        </div>
      </button>
      <ul
        className="o-dimension-section-bar_dimension-details"
      >
        <li
          className="h6"
        >
          6 criteria
        </li>
        <li
          className="h6"
        >
          est. time 60 min.
        </li>
      </ul>
    </li>
    <li
      className="o-dimension-section-bar_dimension"
    >
      <button
        className="o-dimension-section-bar_button UTILITY"
        onClick={[Function]}
      >
        <div
          class="o-dimension-section-bar_selector"
        >
          <span
            className="o-dimension-section-bar_selector-title"
          >
            <span
              className="a-icon a-icon__large"
            >
              <svg
                class="cf-icon-svg"
                viewBox="0 0 834 1200"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M804.4 505.7L746 495.6c-3.2-.6-6.5-.6-9.7 0-7.8-30.4-19.8-59.5-35.9-86.5 2.7-1.9 5-4.2 6.9-6.9l34.2-48.5c9.5-13.5 7.8-34.1-3.9-45.7l-51.4-51.4c-11.7-11.7-32.2-13.4-45.7-3.9L592 286.9c-2.7 1.9-5 4.2-6.9 6.9-27-16-56.1-28.1-86.5-35.9.6-3.2.6-6.5 0-9.7l-10.1-58.4c-2.8-16.3-18.6-29.6-35.1-29.6h-72.7c-16.5 0-32.3 13.3-35.1 29.6l-10.1 58.4c-.6 3.2-.6 6.5 0 9.7-30.4 7.8-59.5 19.8-86.5 35.9-1.9-2.7-4.2-5-6.9-6.9l-48.5-34.2c-13.5-9.5-34.1-7.8-45.7 3.9L96.5 308c-11.7 11.7-13.4 32.2-3.9 45.7l34.2 48.5c1.9 2.7 4.2 5 6.9 6.9-16 27-28.1 56.1-35.9 86.5-3.2-.6-6.5-.6-9.7 0l-58.4 10.1C13.3 508.5 0 524.3 0 540.8v72.7c0 16.5 13.3 32.3 29.6 35.1L88 658.8c3.2.6 6.5.6 9.7 0 7.8 30.4 19.8 59.5 35.9 86.5-2.7 1.9-5 4.2-6.9 6.9l-34.2 48.5c-9.5 13.5-7.8 34.1 3.9 45.7l51.4 51.4c11.7 11.7 32.2 13.4 45.7 3.9l48.5-34.2c2.7-1.9 5-4.2 6.9-6.9 27 16 56.1 28.1 86.5 35.9-.6 3.2-.6 6.5 0 9.7l10.1 58.4c2.8 16.3 18.6 29.6 35.1 29.6h72.7c16.5 0 32.3-13.3 35.1-29.6l10.1-58.4c.6-3.2.6-6.5 0-9.7 30.4-7.8 59.5-19.8 86.5-35.9 1.9 2.7 4.2 5 6.9 6.9l48.5 34.2c13.5 9.5 34.1 7.8 45.7-3.9l51.4-51.4c11.7-11.7 13.4-32.2 3.9-45.7l-34.2-48.5c-1.9-2.7-4.2-5-6.9-6.9 16-27 28.1-56.1 35.9-86.5 3.2.6 6.5.6 9.7 0l58.4-10.1c16.3-2.8 29.6-18.6 29.6-35.1v-72.7c.1-16.6-13.2-32.4-29.5-35.2zM417 719.2c-78.4 0-142-63.6-142-142s63.6-142 142-142 142 63.6 142 142-63.6 142-142 142z"
                />
              </svg>
            </span>
             
            Utility
             
            <span
              className="o-dimension-section-bar_selector-status"
            >
               
            </span>
          </span>
          <span
            className="o-dimension-section-bar_selector-text"
          >
            Supports effective teaching
          </span>
        </div>
      </button>
      <ul
        className="o-dimension-section-bar_dimension-details"
      >
        <li
          className="h6"
        >
          5 criteria
        </li>
        <li
          className="h6"
        >
          est. time 60 min.
        </li>
      </ul>
    </li>
    <li
      className="o-dimension-section-bar_dimension"
    >
      <button
        className="o-dimension-section-bar_button QUALITY"
        onClick={[Function]}
      >
        <div
          class="o-dimension-section-bar_selector"
        >
          <span
            className="o-dimension-section-bar_selector-title"
          >
            <span
              className="a-icon a-icon__large"
            >
              <svg
                class="cf-icon-svg"
                viewBox="0 0 969.7 1200"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M843.8 592.5l94.6-87.5c52.5-48.5 37.3-95.1-33.7-103.6l-128-15.2-129-15.3-81.5-176.6-27-58.5c-30-64.9-79-64.9-108.9 0l-27 58.5L322 371l-129.1 15.3-128 15.2c-71 8.4-86.1 55-33.6 103.6l47.3 43.7 47.3 43.7 95.4 88.3L196 808.2l-12.6 63.2-12.6 63.2c-13.9 70.1 25.7 98.9 88.1 64l225.9-126.4 225.9 126.4c62.4 34.9 102 6.1 88.1-64l-12.6-63.2-37.9-190.7 95.5-88.2z"
                />
              </svg>
            </span>
             
            Quality
             
            <span
              className="o-dimension-section-bar_selector-status"
            >
               
            </span>
          </span>
          <span
            className="o-dimension-section-bar_selector-text"
          >
            Presents clear, accurate, and objective information
          </span>
        </div>
      </button>
      <ul
        className="o-dimension-section-bar_dimension-details"
      >
        <li
          className="h6"
        >
          4 criteria
        </li>
        <li
          className="h6"
        >
          est. time 30 min.
        </li>
      </ul>
    </li>
    <li
      className="o-dimension-section-bar_dimension"
    >
      <button
        className="o-dimension-section-bar_button EFFICACY"
        onClick={[Function]}
      >
        <div
          class="o-dimension-section-bar_selector"
        >
          <span
            className="o-dimension-section-bar_selector-title"
          >
            <span
              className="a-icon a-icon__large"
            >
              <svg
                class="cf-icon-svg"
                viewBox="0 0 651.7 1200"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M651.7 367.7v-24l-91.4-91.4-66.6-66.6L469 161H35c-19.3.1-34.9 15.7-35 35v817.8c.1 19.3 15.7 34.9 35 35h581.7c19.3-.1 34.9-15.7 35-35V367.7zm-60 59v562.1H60V221h359.6v134.8c.1 20 16.2 36.2 36.2 36.2h135.9v34.7z"
                />
                <path
                  d="M279.6 802.8l-77.8-93.4c-14.1-17-39.4-19.3-56.3-5.1-17 14.1-19.3 39.4-5.1 56.3l114.5 137.5c7.6 9.1 18.9 14.4 30.7 14.4 1.1 0 2.3-.1 3.4-.2 13-1.1 24.7-8.5 31.2-19.9l204.7-354.5c11.2-19.1 4.8-43.6-14.3-54.7s-43.6-4.8-54.7 14.3c-.1.2-.2.3-.3.5l-176 304.8z"
                />
              </svg>
            </span>
             
            Efficacy
             
            <span
              className="o-dimension-section-bar_selector-status"
            >
               
            </span>
          </span>
          <span
            className="o-dimension-section-bar_selector-text"
          >
            Has validated impact on knowledge, skills, or behaviors
          </span>
        </div>
      </button>
      <ul
        className="o-dimension-section-bar_dimension-details"
      >
        <li
          className="h6"
        >
          1-3 criteria
        </li>
        <li
          className="h6"
        >
          est. time 30 min.
        </li>
      </ul>
    </li>
  </ul>
  <span />
  <div />
  <div
    className="block block__flush-bottom block__padded-top block__border-top"
  >
    <div
      className="m-btn-group m-btn-group__wide"
    >
      <button
        className="a-btn"
        disabled={true}
      >
        Continue to summary
      </button>
      <button
        className="a-btn a-btn__link"
        onClick={[Function]}
      >
        Start over with a new review
      </button>
      <div
        aria-describedby="modal-start-over_desc"
        aria-hidden="true"
        aria-labelledby="modal-start-over_title"
        className="o-modal"
        id="modal-start-over"
        role="alertdialog"
      >
        <div
          className="o-modal_backdrop"
        />
        <div
          className="o-modal_container"
        >
          <form
            className="o-modal_content"
          >
            <span>
              <div
                className="o-modal_body"
              >
                <button
                  className="o-modal_close a-btn a-btn__link"
                  onClick={[Function]}
                >
                  Close  
                  <span
                    className="a-icon a-icon__large"
                  >
                    <svg
                      className="cf-icon-svg"
                      viewBox="0 0 1000 1200"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M500 105.2c-276.1 0-500 223.9-500 500s223.9 500 500 500 500-223.9 500-500-223.9-500-500-500zm261.8 692.2c19.4 19.6 19.3 51.3-.3 70.7-19.5 19.3-50.9 19.3-70.4 0L499.6 676.6 308 868.1c-19.6 19.4-51.3 19.3-70.7-.3-19.3-19.5-19.3-50.9 0-70.4l191.6-191.5-191.6-191.6c-19.3-19.8-18.9-51.4.9-70.7 19.4-18.9 50.4-18.9 69.8 0l191.6 191.5 191.5-191.5c19.6-19.4 51.3-19.3 70.7.3 19.3 19.5 19.3 50.9 0 70.4L570.3 605.9l191.5 191.5z"
                      />
                    </svg>
                  </span>
                </button>
                <h1
                  id="modal-start-over_title"
                >
                  Starting over
                </h1>
                <div
                  id="modal-start-over_desc"
                >
                  <p>
                    Starting a new review will erase your answers for all dimensions. Are you sure you want to start a new review?
                  </p>
                </div>
              </div>
              <div
                className="o-modal_footer"
              >
                <div
                  className="m-btn-group"
                >
                  <button
                    className="a-btn"
                    formAction="../../tdp/crt-start/"
                    onClick={[Function]}
                  >
                    Yes
                  </button>
                  <button
                    className="a-btn a-btn__link"
                    onClick={[Function]}
                  >
                    No, return to current review
                  </button>
                </div>
              </div>
            </span>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
